pkg/login/social/okta_oauth.go

```
@@ -0,0 +1,126 @@
+	Name              string `json:"name"`
+}
+
+type OktaUserInfoJson struct {
+	Name        string              `json:"name"`
+	DisplayName string              `json:"display_name"`
+	Login       string              `json:"login"`
+	Username    string              `json:"username"`
+	Email       string              `json:"email"`
+	Upn         string              `json:"upn"`
+	Attributes  map[string][]string `json:"attributes"`
+	Groups      []string            `json:"groups"`
+	rawJSON     []byte
+}
+
+func (s *SocialOkta) Type() int {
+	return int(models.OKTA)
+}
+
+func (s *SocialOkta) UserInfo(client *http.Client, token *oauth2.Token) (*BasicUserInfo, error) {
+	idToken := token.Extra("id_token")
+	if idToken == nil {
+		return nil, fmt.Errorf("No id_token found")
+	}
+
+	parsedToken, err := jwt.ParseSigned(idToken.(string))
+	if err != nil {
+		return nil, fmt.Errorf("Error parsing id token")
+	}
+
+	var claims OktaClaims
+	if err := parsedToken.UnsafeClaimsWithoutVerification(&claims); err != nil {
+		return nil, fmt.Errorf("Error getting claims from id token")
+	}
+
+	email := claims.extractEmail()
+
+	if email == "" {
+		return nil, errors.New("Error getting user info: No email found in access token")
+	}
+
+	var data OktaUserInfoJson
+	s.extractAPI(&data, client)
+	role := s.extractRole(&data)
+	groups := s.GetGroups(&data)
+
+	return &BasicUserInfo{
+		Id:     claims.ID,
+		Name:   claims.Name,
+		Email:  email,
+		Login:  email,
+		Role:   role,
+		Groups: groups,
+	}, nil
+}
+
+func (s *SocialOkta) extractAPI(data *OktaUserInfoJson, client *http.Client) bool {
+	rawUserInfoResponse, err := HttpGet(client, s.apiUrl)
+	if err != nil {
+		s.log.Debug("Error getting user info response", "url", s.apiUrl, "error", err)
+		return false
+	}
+	data.rawJSON = rawUserInfoResponse.Body
+
+	err = json.Unmarshal(data.rawJSON, data)
+	if err != nil {
+		s.log.Error("Error decoding user info response", "raw_json", data.rawJSON, "error", err)
```

I think you should propagate the error here to the caller, instead of swallowing it.